pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -172,16 +172,20 @@ func (e *cloudWatchExecutor) newSession(region string, pluginCtx backend.PluginC
 		region = dsInfo.region
 	}
 
-	return e.sessions.GetSession(region, awsds.AWSDatasourceSettings{
-		Profile:       dsInfo.profile,
-		Region:        region,
-		AuthType:      dsInfo.authType,
-		AssumeRoleARN: dsInfo.assumeRoleARN,
-		ExternalID:    dsInfo.externalID,
-		Endpoint:      dsInfo.endpoint,
-		DefaultRegion: dsInfo.region,
-		AccessKey:     dsInfo.accessKey,
-		SecretKey:     dsInfo.secretKey,
+	return e.sessions.GetSession(awsds.SessionConfig{
+		Config: *pluginCtx.DataSourceInstanceSettings,
```

Comment:
Rather than providing this we probably want to provide a HTTP client so that we can make use of the configured `HTTPClientProvider` in Grafana.

Now, your doing `httpclient.New` [here](https://github.com/grafana/grafana-aws-sdk/blob/b03a5d9c98fbdfb3a35614c802610c43a0c26e68/pkg/awsds/sessions.go#L159) which will not reuse the Grafana configured `HTTPClientProvider`.

Reply:
I don't think CloudWatch currently uses it, but google cloud monitoring is an example of a datasource that uses an http client.Provider. If I'm understanding correctly the idea is to take a httpclient.Provider in ProvideService() and use that to create clients for the sdk.

Reply:
@iwysiu correct :+1: 

> take a httpclient.Provider in ProvideService()

https://github.com/grafana/grafana/blob/7422789ec7ab665fd5d01e3b891f44ac427061c5/pkg/tsdb/cloudmonitoring/cloudmonitoring.go#L74

> use that to create clients for the sdk

https://github.com/grafana/grafana/blob/7422789ec7ab665fd5d01e3b891f44ac427061c5/pkg/tsdb/cloudmonitoring/cloudmonitoring.go#L212
https://github.com/grafana/grafana/blob/7422789ec7ab665fd5d01e3b891f44ac427061c5/pkg/tsdb/cloudmonitoring/httpclient.go#L60

Graphite might be an easier example:
https://github.com/grafana/grafana/blob/7422789ec7ab665fd5d01e3b891f44ac427061c5/pkg/tsdb/graphite/graphite.go#L70-L77

Reply:
thanks @marefr. That solves it for core plugins, for external plugins, shall we use the HTTPClientProvider exposed by the SDK?

https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/httpclient/provider.go#L50

Reply:
Yeah, you can do that. Right now there's no middleware per default when creating a new provided in the SDK so wouldn't currently provide any benefit compared to current using AWS SDK builtin httpclient.

Reply:
Got it :+1:  Can you take a look to this PR again?